cordial_behavior/scripts/behaviors.py

```diff
@@ -0,0 +1,260 @@
+from threading import Timer
+import os
+
+RATE_FEEDBACK =""
+#data = "[{'start': 0.075, 'time': 2,'type': 'action', 'id': 'QT/bye'},{'start': 0.075, 'time': 2,'type': 'action', 'id': 'QT/point_front'}, {'start': 0.075,'time': 2, 'type': 'viseme', 'id': 'POSTALVEOLAR'},{'start': 0.006, 'time': 2,  'type': 'action', 'id': 'happy_face'}]"
+TTS_MESSAGE = ''
+SPEAKER_DONE = False
+TRACKER_STATE = ''
+TRACKER_DONE = False
+DETECTOR_MESSAGE = ''
+
+
+FEEDBACK_MESSAGE = ''
+INTERACTION_MESSAGE = ''
+INTERACTION_CONTINUE = True
+
+class LongBehaviorServer():
+	_feedback = InteractionFeedback()
+	_result = InteractionResult()
+
+	def __init__(self, name):
+		self.action_name = name
+		self.action = actionlib.SimpleActionServer(self.action_name, InteractionAction, self.execute_goal, False)
+		self.action.start()
+
+	def execute_goal(self, goal):
+		global DETECTOR_MESSAGE , DETECTOR_MESSAGE, TRACKER_STATE
+		goal_name = goal.interacting_action
+		success = True
+		if goal.optional_data != '':
+			DETECTOR_MESSAGE = goal.optional_data
+		bm = BehaviorManager()
+		bm.handle_tracker(DETECTOR_MESSAGE)
+		self._feedback.interacting_action = goal_name
+		self._feedback.interaction_state = FEEDBACK_MESSAGE
+		## Decide when to send the feedback
+		RATE_FEEDBACK = rospy.Rate(10)
+		while not TRACKER_DONE:
+			if self.action.is_preempt_requested():
+					self.action.set_preempted()
+					success = False
+			self.action.publish_feedback(self._feedback)
+			rospy.Rate(10)
+
+		if success:
+			self._result.interaction_continue = INTERACTION_CONTINUE
+			self._result.interacting_action = goal_name
+			self._result.message = INTERACTION_MESSAGE
+			TRACKER_STATE = ''
+			TRACKER_DONE = False
+			DETECTOR_MESSAGE = ''
+			self.action.set_succeeded(self._result)
+
+
+class BehaviorServer():
+	_feedback = InteractionFeedback()
+	_result = InteractionResult()
+	def __init__(self, name):
+		self.action_name = name
+		self.action = actionlib.SimpleActionServer(self.action_name, InteractionAction, self.execute_goal, False)
+		self.action.start()
+
+	def execute_goal(self, goal):
+		global TTS_MESSAGE, SPEAKER_DONE
+		goal_name = goal.interacting_action
+		success = True
+		if goal.optional_data != '':
+			TTS_MESSAGE = goal.optional_data
+		bm = BehaviorManager()
+		bm.handle_behavior(TTS_MESSAGE)
+		self._feedback.interacting_action = goal_name
+		self._feedback.interaction_state = FEEDBACK_MESSAGE
+		## Decide when to send the feedback
+		# self.action.publish_feedback(self._feedback)
+		while not SPEAKER_DONE:
+			if self.action.is_preempt_requested():
+					self.action.set_preempted()
+					success = False
+			rospy.Rate(10)
+		if success:
+			self._result.interaction_continue = INTERACTION_CONTINUE
+			self._result.interacting_action = goal_name
+			self._result.message = INTERACTION_MESSAGE
+			SPEAKER_DONE = False
+			TTS_MESSAGE = ''
+			self.action.set_succeeded(self._result)
+		
+
+class BehaviorManager():
+
+	def __init__(self):
```

Comment:
Does this handle having multiple robots, or is there one behavior manager per robot?